content/blog/2020-03-06-cmake-improvements.md

```diff
@@ -0,0 +1,50 @@
+---
+author: Zack Galbreath
+author-link: https://github.com/zackgalbreath
+date: "2020-03-06T00:00:00Z"
+published: true
+title: Improvements to gRPC’s CMake Buildsystem
+url: blog/cmake-improvements
+---
+
+For the past few months, [Kitware Inc.](https://www.kitware.com/) has been working with the gRPC team to improve gRPC’s CMake support. The goal of the effort was to modernize gRPC’s CMake build with the most current features and techniques CMake has to offer. This has improved the user experience for gRPC developers choosing to use gRPC’s CMake as a build system. During the effort the CMake build was looked at as a whole, and CMake related issues in GitHub were explored and resolved. A number of improvements were made that will give developers and end users a better experience when building gRPC with CMake.
+
+One of the more exciting changes is the ability to seamlessly add gRPC to any CMake project and have all of its dependent libraries built using a simple CMake file. Prior to our recent changes this was a multi-step process. The user had to build and install each of gRPC’s dependencies separately, then build and install gRPC before finally building their own project. Now, this can all be done in one step. The following CMake code clones and builds the latest stable release of gRPC as documented [here](https://github.com/grpc/grpc/blob/master/src/cpp/README.md#cmake):
+
+```cmake
+cmake_minimum_required(VERSION 3.15)
+project(my_exe)
+
+include(FetchContent)
+
+FetchContent_Declare(
+  gRPC
+  GIT_REPOSITORY https://github.com/grpc/grpc
+  GIT_TAG        {{< param grpc_release_tag >}}
```

Comment:
Like all blogposts this one will be timestamped so, IMHO, the GIT_TAG value should be fixed to what it is at the time of writing.

Given that you use 1.27.0 below (in `libgrpc++.so.1.27.0-dev`), maybe use the same version here.

Reply:
Actually, given that this will be available as of v1.28.0, why not use 1.28.0?

Reply:
ok, let's set this to v1.28.0

Reply:
@lucperkins - can you make this change along with the other updates you'll be making to the page?